test/widgets/compose_box_test.dart

```diff
@@ -195,15 +220,181 @@ void main() {
+        [eg.otherUser.userId], selfUserId: eg.selfUser.userId);
+      await prepareComposeBox(tester, narrow: narrow);
+
+      await checkStartTyping(tester, narrow);
+
+      connection.prepare(json: {});
+      await tester.pump(store.typingNotifier.typingStoppedWaitPeriod);
+      checkTypingRequest(TypingOp.stop, narrow);
+    });
+
+    testWidgets('smoke ChannelNarrow', (tester) async {
+      const narrow = ChannelNarrow(123);
+      final destinationNarrow = TopicNarrow(narrow.streamId, 'test topic');
+      await prepareComposeBox(
+        tester, narrow: narrow, topic: destinationNarrow.topic);
+
+      await checkStartTyping(tester, destinationNarrow);
+
+      connection.prepare(json: {});
+      await tester.pump(store.typingNotifier.typingStoppedWaitPeriod);
+      checkTypingRequest(TypingOp.stop, destinationNarrow);
+    });
+
+    testWidgets('clearing text sends a "typing stopped" notice', (tester) async {
+      await prepareComposeBox(tester, narrow: narrow);
+
+      await checkStartTyping(tester, narrow);
+
+      connection.prepare(json: {});
+      await tester.enterText(contentInputFinder, '');
+      checkTypingRequest(TypingOp.stop, narrow);
+    });
+
+    testWidgets('hitting send button sends a "typing stopped" notice', (tester) async {
+      await prepareComposeBox(tester, narrow: narrow);
+
+      await checkStartTyping(tester, narrow);
+
+      connection.prepare(json: {});
+      connection.prepare(json: SendMessageResult(id: 123).toJson());
+      await tester.tap(find.byIcon(Icons.send));
+      await tester.pump(Duration.zero);
+      final requests = connection.takeRequests();
+      checkSetTypingStatusRequests([requests.first], [(TypingOp.stop, narrow)]);
+      check(requests).length.equals(2);
+    });
+
+    Future<void> prepareComposeBoxWithNavigation(WidgetTester tester) async {
+      addTearDown(testBinding.reset);
+      await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot());
+
+      store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
+      connection = store.connection as FakeApiConnection;
+
+      await tester.pumpWidget(const ZulipApp());
+      await tester.pump();
+      final navigator = await ZulipApp.navigator;
+      unawaited(navigator.push(MaterialAccountWidgetRoute(
+        accountId: eg.selfAccount.id, page: const ComposeBox(narrow: narrow))));
+      await tester.pumpAndSettle();
+    }
+
+    testWidgets('navigating away sends a "typing stopped" notice', (tester) async {
+      await prepareComposeBoxWithNavigation(tester);
+
+      await checkStartTyping(tester, narrow);
+
+      connection.prepare(json: {});
+      (await ZulipApp.navigator).pop();
+      await tester.pump(Duration.zero);
+      checkTypingRequest(TypingOp.stop, narrow);
+    });
+
+    testWidgets('for content input, unfocusing sends a "typing stopped" notice '
+                'and refocusing sends a "typing started" notice', (tester) async {
```

Comment:
> refocusing sends a "typing started" notice

Is this desirable? It seems undesirable to me.

I think I see why it happens in the current implementation. It's not a big deal and we don't have to fix it before merge. But as I said in the previous thread:
https://github.com/zulip/zulip-flutter/pull/897#discussion_r1818188643
the test should focus on testing behaviors we actually want. Moreover if we do have a test that checks behavior we don't want, it should be very clearly labeled as such — otherwise, the test becomes an extra barrier in the way of us fixing the behavior.